tests/release/e2e-test-push-image-to-pyxis.go

```diff
@@ -147,68 +159,128 @@ var _ = framework.ReleaseSuiteDescribe("[HACBS-1571]test-release-e2e-push-image-
 					return false
 				}
 
-				return prList.Items[0].HasStarted() && prList.Items[0].IsDone() && prList.Items[0].Status.GetCondition(apis.ConditionSucceeded).IsTrue()
+				return releasePr.HasStarted() && releasePr.IsDone() && releasePr.Status.GetCondition(apis.ConditionSucceeded).IsTrue() &&
+					additionalRleasePr.HasStarted() && additionalRleasePr.IsDone() && additionalRleasePr.Status.GetCondition(apis.ConditionSucceeded).IsTrue()
 			}, releasePipelineRunCompletionTimeout, defaultInterval).Should(BeTrue())
 		})
 
-		It("validate the result of task create-pyxis-image contains id and succeeded.", func() {
+		It("validate the result of task create-pyxis-image contains image ids.", func() {
 			Eventually(func() bool {
-				prList, err := fw.AsKubeAdmin.TektonController.ListAllPipelineRuns(managedNamespace)
-				if prList == nil || err != nil || len(prList.Items) < 1 {
+
+				releasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(releasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting Release PirpelineRun %s:\n %s", releasePr, err)
+					return false
+				}
+				additionalRleasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(additionalReleasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting PirpelineRun %s:\n %s", additionalRleasePr, err)
 					return false
 				}
+				re := regexp.MustCompile("[a-fA-F0-9]{24}")
 
-				pr, err := kubeController.Tektonctrl.GetPipelineRun(prList.Items[0].Name, managedNamespace)
-				Expect(err).NotTo(HaveOccurred())
+				trReleasePr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), releasePr, "create-pyxis-image")
+				if err != nil {
+					Expect(err).NotTo(HaveOccurred())
+				}
 
-				tr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), pr, "create-pyxis-image")
-				Expect(err).NotTo(HaveOccurred())
+				trAdditionalReleasePr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), additionalRleasePr, "create-pyxis-image")
+				if err != nil {
+					Expect(err).NotTo(HaveOccurred())
+				}
 
-				re := regexp.MustCompile("^[0-9a-z]{10,50}")
-				returnedImageID := re.FindString(tr.Status.TaskRunResults[0].Value.StringVal)
+				if len(re.FindAllString(trReleasePr.Status.TaskRunResults[0].Value.StringVal, -1)) >
+					len(re.FindAllString(trAdditionalReleasePr.Status.TaskRunResults[0].Value.StringVal, -1)) {
+					imageIDs = re.FindAllString(trReleasePr.Status.TaskRunResults[0].Value.StringVal, -1)
+				} else {
+					imageIDs = re.FindAllString(trAdditionalReleasePr.Status.TaskRunResults[0].Value.StringVal, -1)
+				}
 
-				return strings.Contains(string(tr.Status.Status.Conditions[0].Status), "True") &&
-					strings.Contains(string(tr.Status.TaskRunResults[0].Name), "containerImageIDs") && len(returnedImageID) > 10
+				return len(imageIDs[0]) > 10 && len(imageIDs[1]) > 10
```

Comment:
where is the value `10` coming from? should it be added to some constant

Reply:
imo there should be an assertion above this that `len(imageIDs) == 2`

Reply:
@mmalina ,  according to my understand that the imageIDs related to SnapShots release-service detects , 
can be any case that we imageIDs  not two  ?

Reply:
If this is the correct release pipeline with the two components, then we expect there to be two imageIDs as a result - and if not, something went wrong. But I agree with Pavel - the test should check that first. Because if for some reason there is only one imageID, your current code would fail on something like "array index out of bounds" on `imageIDs[1]` which is much less nice for debugging.